zerver/lib/soft_deactivation.py

```diff
@@ -158,8 +159,15 @@ def add_missing_messages(user_profile: UserProfile) -> None:
         user_profile, stream_messages, all_stream_subscription_logs)
 
     # Doing a bulk create for all the UserMessage objects stored for creation.
-    if len(user_messages_to_insert) > 0:
-        UserMessage.objects.bulk_create(user_messages_to_insert)
+    while len(user_messages_to_insert) > 0:
+        messages, user_messages_to_insert = (
+            user_messages_to_insert[0:BULK_CREATE_BATCH_SIZE],
+            user_messages_to_insert[BULK_CREATE_BATCH_SIZE:])
+        UserMessage.objects.bulk_create(messages)
+        # Update last_active_message_id if there are more messages to create
```

Comment:
Why don't we do this unconditionally?

Reply:
Yes, we should do this unconditionally. I thought the last update was being done by the callers of this code, but it isn't. In any case, it would be better to do this unconditionally, here.